patches/amd-mainline/Tensile/0002-Capture-configure-time-PATH-when-invoking-tensile-at.patch

```diff
@@ -1,31 +1,43 @@
-From 560f1a9f4b725a9b9d0299249b44c19242feddf0 Mon Sep 17 00:00:00 2001
+From 106bf269597b04a3779349b3683005326f66bc1c Mon Sep 17 00:00:00 2001
 From: Stella Laurenzo <stellaraccident@gmail.com>
 Date: Fri, 14 Feb 2025 16:52:43 -0800
-Subject: [PATCH 2/2] Capture configure-time PATH when invoking tensile at
+Subject: [PATCH 2/3] Capture configure-time PATH when invoking tensile at
  build time.
 
 ---
- Tensile/cmake/TensileConfig.cmake | 9 ++++++++-
- 1 file changed, 8 insertions(+), 1 deletion(-)
+ Tensile/cmake/TensileConfig.cmake | 29 +++++++++++++++++++++++++----
+ 1 file changed, 25 insertions(+), 4 deletions(-)
 
 diff --git a/Tensile/cmake/TensileConfig.cmake b/Tensile/cmake/TensileConfig.cmake
-index e5d0f62a..c2312c14 100644
+index e5d0f62a..dfd848f1 100644
 --- a/Tensile/cmake/TensileConfig.cmake
 +++ b/Tensile/cmake/TensileConfig.cmake
-@@ -219,6 +219,12 @@ function(TensileCreateLibraryFiles
+@@ -215,10 +215,24 @@ function(TensileCreateLibraryFiles
+     set(Options ${Options} "--architecture=${archString}")
+   endif()
+ 
++  # We do not need to do device enumeration at library build time.
++  set(Options ${Options} "--no-enumerate")
++
+   set(CommandLine ${Script} ${Options} ${Tensile_LOGIC_PATH} ${Tensile_OUTPUT_PATH} HIP)
    if (WIN32 OR (VIRTUALENV_BIN_DIR AND VIRTUALENV_PYTHON_EXENAME))
      set(CommandLine ${VIRTUALENV_BIN_DIR}/${VIRTUALENV_PYTHON_EXENAME} ${CommandLine})
    endif()
 +  # Tensile relies on the tools from the path, so capture the configure time 
 +  # path. It would be better if this were explicit, but that would be a pretty 
 +  # big change.
++  set(ESC_PATH "$ENV{PATH}")
++  if(WIN32)
++    string(REPLACE ";" "$<SEMICOLON>" ESC_PATH "${ESC_PATH}")
++  endif()
++  set(ENV_PATH_ARG "PATH=${ESC_PATH}")
 +  set(CommandLine 
-+    "${CMAKE_COMMAND}" -E env "'PATH=$ENV{PATH}'" --
++    "${CMAKE_COMMAND}" -E env "PATH=${ESC_PATH}" --
 +    ${CommandLine})
```

Comment:
You might be able to use `file(TO_CMAKE_PATH` and/or `file(TO_NATIVE_PATH` instead of `string(REPLACE`: https://cmake.org/cmake/help/latest/command/file.html#to-cmake-path . Not sure about what specifically `-E env PATH=` is expecting.

Reply:
I'm going to leave it like this. Will normalize when upstreaming. This probably isn't the last time we're going to amend this patch.